[ja]Update /content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md

ziyi-xie opened the pull request:
There are several updates in [the original English text](https://github.com/kubernetes/website/blob/main/content/en/docs/concepts/scheduling-eviction/assign-pod-node.md) and therefore this page is outdated.
This PR updated Japanese translation for the [/content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md](https://kubernetes.io/ja/docs/concepts/scheduling-eviction/assign-pod-node/) file.

Fixes https://github.com/kubernetes/website/issues/38756

ziyi-xie commented:
/retitle [WIP][ja]Update /content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md

ziyi-xie commented:
/assign @nasa9084
/cc @atoato88 @t-inu
Hi! I've updated Japanese translation for the [/content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md](https://kubernetes.io/ja/docs/concepts/scheduling-eviction/assign-pod-node/) file line 1 to line 157 (That would be line 1 to line 112 in the Japanese version), would you mind checking if everything looks alright at your convenience time please? Here is [the original English text](https://github.com/kubernetes/website/blob/main/content/en/docs/concepts/scheduling-eviction/assign-pod-node.md) and thank you so much!

ziyi-xie commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -1,138 +1,115 @@
+詳しい情報については[Podをノードに割り当てる](/ja/docs/tasks/configure-pod-container/assign-pods-nodes/)を参照してください。
 
-## 補足: ビルトインNodeラベル {#built-in-node-labels}
+## Affinityと反Affinity {#affinity-and-anti-affinity}
```
Here's the original English test:

> Affinity and anti-affinity

And I'm a little confused about the translation of them over here, could you tell me which of the following is better? 
I'd appreciate it if you could give me some advice about that and thank you so much!

- Affinityと反Affinity
- アフィニティとアンチアフィニティ
- アフィニティと反アフィニティ

atoato88 replied:
I greped which translation is used on other ja pages and `アンチアフィニティ` is used on other pages.
And I usualy use the word of `アンチアフィニティ` in talking with other japanese speakers.

I suggest using `アンチアフィニティ`.

ziyi-xie replied:
Thank you for your comment! I’ll modify all of them to `アンチアフィニティ` in the next  commit!

ziyi-xie replied:
For the word `affinity`, it's translated as `Affinity` in  the following documents,

- https://kubernetes.io/ja/docs/tasks/configure-pod-container/assign-pods-nodes-using-node-affinity/
- https://kubernetes.io/ja/docs/concepts/scheduling-eviction/kube-scheduler/

and `アフィニティ` in these following link. 

- https://kubernetes.io/ja/docs/concepts/scheduling-eviction/taint-and-toleration/
- https://kubernetes.io/ja/docs/concepts/storage/persistent-volumes/　　　
(The word `Node Affinity` is actually translated as `ノードアフィニティ` in this link.)

I think we should unify our translation standard.

ziyi-xie commented:
/remove-language en

ziyi-xie commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -1,138 +1,115 @@
-この例ではオペレーター`In`が使われています。
-Nodeアフィニティでは、`In`、`NotIn`、`Exists`、`DoesNotExist`、`Gt`、`Lt`のオペレーターが使用できます。
-`NotIn`と`DoesNotExist`はNodeアンチアフィニティ、またはPodを特定のNodeにスケジュールさせない場合に使われる[Taints](/ja/docs/concepts/scheduling-eviction/taint-and-toleration/)に使用します。
+`nodeSelectorTerms`に複数の条件を指定し、`nodeAffinity`タイプも指定した場合、Podは指定した条件のいずれかを満たしたNodeへスケジュールされます(条件はORされます)。
```
Here's the original English text:

> If you specify multiple terms in nodeSelectorTerms associated with nodeAffinity types, then the Pod can be scheduled onto a node if one of the specified terms can be satisfied (terms are ORed).

I'm a little confused about the translation of 'associated with' over here, is it means 'と一緒に' or '関連する' ?
I'd appreciate it if you could give me some advice about that and thank you so much!

nasa9084 replied:
I think the current translation is good~

ziyi-xie replied:
Thank you for your kind advice! Then I'll keep the current translation.

t-inu reviewed (commented):
I'm still in the middle of my review, but I left comments.

atoato88 commented:
The target page on preview site for this PR:
- https://deploy-preview-38800--kubernetes-io-main-staging.netlify.app/ja/docs/concepts/scheduling-eviction/assign-pod-node/

ziyi-xie commented:
> The target page on preview site for this PR:
> 
> * https://deploy-preview-38800--kubernetes-io-main-staging.netlify.app/ja/docs/concepts/scheduling-eviction/assign-pod-node/

Thank you so much for the link!

atoato88 commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -1,138 +1,115 @@
+## Nodeラベル {#built-in-node-labels}
 
-### ステップ0: 前提条件
+他の多くのKubernetesオブジェクトと同様に、Nodeにも[ラベル](/ja/docs/concepts/overview/working-with-objects/labels/)があります。[手動でラベルを貼り付ける](/ja/docs/tasks/configure-pod-container/assign-pods-nodes/#ラベルをノードに追加する)ことができます。
```
```suggestion
他の多くのKubernetesオブジェクトと同様に、Nodeにも[ラベル](/ja/docs/concepts/overview/working-with-objects/labels/)があります。[手動でラベルを付ける](/ja/docs/tasks/configure-pod-container/assign-pods-nodes/#ラベルをノードに追加する)ことができます。
```
I think `ラベルをつける` is better.

atoato88 commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -1,138 +1,115 @@
 
-### ステップ0: 前提条件
+他の多くのKubernetesオブジェクトと同様に、Nodeにも[ラベル](/ja/docs/concepts/overview/working-with-objects/labels/)があります。[手動でラベルを貼り付ける](/ja/docs/tasks/configure-pod-container/assign-pods-nodes/#ラベルをノードに追加する)ことができます。
+また、Kubernetesはクラスター内のすべてのNodeに対し、いくつかの標準ラベルを貼り付けます。Nodeラベルの一覧については[よく使われるラベル、アノテーションとtaint](/docs/reference/labels-annotations-taints/)を参照してください。
```
```suggestion
また、Kubernetesはクラスター内のすべてのNodeに対し、いくつかの標準ラベルを付けます。Nodeラベルの一覧については[よく使われるラベル、アノテーションとtaint](/docs/reference/labels-annotations-taints/)を参照してください。
```
ditto.

atoato88 commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -1,138 +1,115 @@
 
-`kubectl get nodes --show-labels`によって、ノードにラベルが付与されたかを確認することができます。
-また、`kubectl describe node "nodename"`から、そのNodeの全てのラベルを表示することもできます。
+ Node分離するのにラベルを使用する場合、{{<glossary_tooltip text="kubelet" term_id="kubelet">}}が修正できないラベルキーを選択するのがオススメです。
```
```suggestion
 Node分離するのにラベルを使用する場合、{{<glossary_tooltip text="kubelet" term_id="kubelet">}}が修正できないラベルキーを選択するのをお勧めします。
```

atoato88 commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
````diff
@@ -1,138 +1,115 @@
-  - name: nginx
-    image: nginx
-```
+1. [Node認証](/docs/reference/access-authn-authz/node/)を使用していることと、 _enabled_ the `NodeRestriction` アドミッションプラグインを _有効_ になっていることを確認しましょう。
````
```suggestion
1. [Node認可](/docs/reference/access-authn-authz/node/)を使用していることと、 _enabled_ the `NodeRestriction` アドミッションプラグインを _有効_ になっていることを確認します。
```
Original word is `Node authorizer`, so `Node認可` is valid not `Node認証`.

atoato88 reviewed (commented):
I reviewed content till the `### Nodeの分離/制限` section.

t-inu commented:
@ziyi-xie 
The file `pod-with-node-affinity.yaml` has already been updated in PR #39247.
You no longer need to update this file.

ziyi-xie commented:
I've updated Japanese translation for the [/content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md](https://kubernetes.io/ja/docs/concepts/scheduling-eviction/assign-pod-node/) file line 158 to line 329 (That would be line 113 to line 221 in the Japanese version), would you mind checking if everything looks alright at your convenience time please? Here is [the original English text](https://github.com/kubernetes/website/blob/main/content/en/docs/concepts/scheduling-eviction/assign-pod-node.md) and thank you so much!

ziyi-xie commented:
> @ziyi-xie The file `pod-with-node-affinity.yaml` has already been updated in PR #39247. You no longer need to update this file.

Thank you so much for the information! I'll modify it as soon as possible.

ziyi-xie commented:
@t-inu @atoato88 Hi! I've updated Japanese translation for the [/content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md](https://kubernetes.io/ja/docs/concepts/scheduling-eviction/assign-pod-node/) file till the end, would you mind checking if everything looks alright at your convenience time please? Here is [the original English text](https://github.com/kubernetes/website/blob/main/content/en/docs/concepts/scheduling-eviction/assign-pod-node.md) and thank you so much!

ziyi-xie commented:
/retitle [ja]Update /content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md

ziyi-xie commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -1,218 +1,239 @@
-Deploymentには、1つのNodeにレプリカを共存させないために`PodAntiAffinity`を付与しています。
+例えば、3つのNodeで構成されるクラスターを想像してください。クラスターを使用してウェブアプリケーションを実行し、さらにインメモリキャッシュ（Redisなど）を使用します。この例では、ウェブアプリケーションとメモリキャッシュの間のレイテンシーは実用的な範囲並みの低さも想定しています。Pod間アフィニティやアンチアフィニティを使って、ウェブサーバとキャッシュをなるべく同じ場所に配置することもできます。
 
+以下のRedisキャッシュのDeploymentの例では、各レプリカはラベル`app=store`が付与されています。`podAntiAffinity`ルールは、`app=store`ラベルを持つ複数のレプリカを単一Nodeに配置しないよう、スケジューラに指示します。これにより、各キャッシュが別々のNodeに作成されます。
```
```suggestion
以下のRedisキャッシュのDeploymentの例では、各レプリカはラベル`app=store`が付与されています。`podAntiAffinity`ルールは、`app=store`ラベルを持つ複数のレプリカを単一Nodeに配置しないよう、スケジューラーに指示します。これにより、各キャッシュが別々のNodeに作成されます。
```

ziyi-xie commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -244,10 +265,7 @@ spec:
-全てのレプリカが`app=store`のラベルが付与されたPodと同じゾーンで稼働するよう、スケジューラーに設定されます。
-また、それぞれのウェブサーバーは1つのノードで稼働されないことも保証されます。
-
+次の Web サーバーのDeployment例では、`app=web-store`ラベルが付与されたレプリカを作成します。Podアフィニティルールは、各レプリカを、`app=store`ラベルが付与されたPodを持つNodeに配置するようスケジューラに指示します。Podアンチアフィニティルールは、1つのNodeに複数の`app=web-store`サーバーを配置しないようにスケジューラーに指示します。
```
```suggestion
次の Web サーバーのDeployment例では、`app=web-store`ラベルが付与されたレプリカを作成します。Podアフィニティルールは、各レプリカを、`app=store`ラベルが付与されたPodを持つNodeに配置するようスケジューラーに指示します。Podアンチアフィニティルールは、1つのNodeに複数の`app=web-store`サーバーを配置しないようにスケジューラーに指示します。
```

ziyi-xie commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -1,218 +1,239 @@
-`nodeSelector`はPodSpecのフィールドです。これはkey-valueペアのマップを特定します。
-あるノードでPodを稼働させるためには、そのノードがラベルとして指定されたkey-valueペアを保持している必要があります(複数のラベルを保持することも可能です)。
-最も一般的な使用方法は、1つのkey-valueペアを付与する方法です。
+他の多くのKubernetesオブジェクトと同様に、Nodeにも[ラベル](/ja/docs/concepts/overview/working-with-objects/labels/)があります。[手動でラベルを付ける](/ja/docs/tasks/configure-pod-container/assign-pods-nodes/#ラベルをノードに追加する)ことができます。
```
```suggestion
他の多くのKubernetesオブジェクトと同様に、Nodeにも[ラベル](/ja/docs/concepts/overview/working-with-objects/labels/)があります。[手動でラベルを付ける](/ja/docs/tasks/configure-pod-container/assign-pods-nodes/#ラベルをNodeに追加する)ことができます。
```

ziyi-xie commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -1,218 +1,239 @@
+Kubernetesは指定された各ラベルを持つNodeにのみ、Podをスケジューリングします。
 
-nodeSelectorを以下のように追加します:
+詳しい情報については[Podをノードに割り当てる](/ja/docs/tasks/configure-pod-container/assign-pods-nodes/)を参照してください。
```
```suggestion
詳しい情報については[PodをNodeに割り当てる](/ja/docs/tasks/configure-pod-container/assign-pods-nodes/)を参照してください。
```

ziyi-xie commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -1,218 +1,239 @@
+`matchExpressions`フィールド内で複数の条件を指定し、`nodeSelectorTerms`にも一つ条件を指定した場合、Podは全ての条件を満たしたNodeへスケジュールされます（条件はANDされます）。
+{{</note>}}
+
+詳細については[Nodeアフィニティを利用してPodをノードに割り当てる](/ja/docs/tasks/configure-pod-container/assign-pods-nodes-using-node-affinity/)を参照してください。
```
```suggestion
詳細については[Nodeアフィニティを利用してPodをNodeに割り当てる](/ja/docs/tasks/configure-pod-container/assign-pods-nodes-using-node-affinity/)を参照してください。
```

ziyi-xie commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -1,218 +1,239 @@
 
-３つのノードから成るクラスターでは、ウェブアプリケーションはredisのようにインメモリキャッシュを保持しています。
-このような場合、ウェブサーバーは可能な限りキャッシュと共存させることが望ましいです。
+Pod間アフィニティとアンチアフィニティは、ReplicaSet、StatefulSet、Deploymentなどのより高レベルなコレクションと併せて使用するとさらに有用です。これらのルールにより、ワークロードのセットが同じ定義されたトポロジーに併置されるように設定できます。たとえば、2つの関連するPodを同じノードに配置することが好ましいとされています。
```
```suggestion
Pod間アフィニティとアンチアフィニティは、ReplicaSet、StatefulSet、Deploymentなどのより高レベルなコレクションと併せて使用するとさらに有用です。これらのルールにより、ワークロードのセットが同じ定義されたトポロジーに併置されるように設定できます。たとえば、2つの関連するPodを同じNodeに配置することが好ましいとされています。
```

ziyi-xie commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -1,218 +1,239 @@
 
-ラベル`app=store`を付与した3つのレプリカから成るredisのdeploymentを記述したyamlファイルを示します。
-Deploymentには、1つのNodeにレプリカを共存させないために`PodAntiAffinity`を付与しています。
+例えば、3つのNodeで構成されるクラスターを想像してください。クラスターを使用してウェブアプリケーションを実行し、さらにインメモリキャッシュ（Redisなど）を使用します。この例では、ウェブアプリケーションとメモリキャッシュの間のレイテンシーは実用的な範囲並みの低さも想定しています。Pod間アフィニティやアンチアフィニティを使って、ウェブサーバとキャッシュをなるべく同じ場所に配置することもできます。
```
```suggestion
例えば、3つのNodeで構成されるクラスターを想像してください。クラスターを使用してウェブアプリケーションを実行し、さらにインメモリキャッシュ（Redisなど）を使用します。この例では、ウェブアプリケーションとメモリキャッシュの間のレイテンシーは実用的な範囲並みの低さも想定しています。Pod間アフィニティやアンチアフィニティを使って、ウェブサーバーとキャッシュをなるべく同じ場所に配置することもできます。
```

ziyi-xie commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -344,18 +342,20 @@ spec:
 
+## Podトポロジー分散制約 {#pod-topology-spread-constraints}
 
+_トポロジー分散制約_ を使って、リージョン、ゾーン、Nodeなどの障害ドメイン間、または定義したその他のトポロジードメイン間で、クラスタ全体にどのように{{< glossary_tooltip text="Pod" term_id="Pod" >}}を分散させるかを制御することができます。これにより、パフォーマンス、予想される可用性、または全体的な使用率を向上させることができます。
```
```suggestion
_トポロジー分散制約_ を使って、リージョン、ゾーン、Nodeなどの障害ドメイン間、または定義したその他のトポロジードメイン間で、クラスター全体にどのように{{< glossary_tooltip text="Pod" term_id="Pod" >}}を分散させるかを制御することができます。これにより、パフォーマンス、予想される可用性、または全体的な使用率を向上させることができます。
```

ziyi-xie commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -1,218 +1,239 @@
 
+`namespaceSelector`を使用し、ラベルで名前空間の集合に対して検索することによって、名前空間を選択することができます。
+アフィニティ項は`namespaceSelector`と`namespaces`フィールドによって選択された名前空間に適用されます。
+要注意なのは、空の `namespaceSelector` ({})はすべての名前空間にマッチし、nullと空の`namespaces`リストとnullの`namespaceSelector`はルールが定義されているPodの名前空間にマッチします。
```
```suggestion
要注意なのは、空の`namespaceSelector`({})はすべての名前空間にマッチし、nullと空の`namespaces`リストとnullの`namespaceSelector`はルールが定義されているPodの名前空間にマッチします。
```

ziyi-xie commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -1,218 +1,239 @@
 
+* Podアフィニティとアンチアフィニティについて、`requiredDuringSchedulingIgnoredDuringExecution`か`preferredDuringSchedulingIgnoredDuringExecution`を指定すると、空の`topologyKey`フォールドは許可されていません。
+* `requiredDuringSchedulingIgnoredDuringExecution` Podアンチアフィニティルールについて、アドミッションコントローラー`LimitPodHardAntiAffinityTopology`が`topologyKey`を`kubernetes.io/hostname`に制限しています。アドミッションコントローラーを修正または無効化すると、トポロジーのカスタマイズができるようになります。
+You can modify or disable the
```
```suggestion
```

ziyi-xie commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -1,218 +1,239 @@
+* Podアフィニティとアンチアフィニティについて、`requiredDuringSchedulingIgnoredDuringExecution`か`preferredDuringSchedulingIgnoredDuringExecution`を指定すると、空の`topologyKey`フォールドは許可されていません。
+* `requiredDuringSchedulingIgnoredDuringExecution` Podアンチアフィニティルールについて、アドミッションコントローラー`LimitPodHardAntiAffinityTopology`が`topologyKey`を`kubernetes.io/hostname`に制限しています。アドミッションコントローラーを修正または無効化すると、トポロジーのカスタマイズができるようになります。
+You can modify or disable the
+  admission controller if you want to allow custom topologies.
```
```suggestion
```

nasa9084 commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -1,218 +1,237 @@
+ラベル接頭辞をNode分離に利用するため:
 
-### ステップ2: PodへのnodeSelectorフィールドの追加
+1. [Node認可](/docs/reference/access-authn-authz/node/)を使用していることと、 _enabled_ the `NodeRestriction` アドミッションプラグインを _有効_ になっていることを確認します。
```
```suggestion
1. [Node認可](/docs/reference/access-authn-authz/node/)を使用していることと、`NodeRestriction` アドミッションプラグインが _有効_ になっていることを確認します。
```

nasa9084 commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -1,218 +1,237 @@
 {{< note >}}
-Pod間アフィニティとアンチアフィニティは、大規模なクラスター上で使用する際にスケジューリングを非常に遅くする恐れのある多くの処理を要します。
-そのため、数百台以上のNodeから成るクラスターでは使用することを推奨されません。
+[DaemonSetのPodを作成する](/ja/docs/concepts/workloads/controllers/daemonset/#デフォルトスケジューラーによってスケジューリングされる場合)DaemonSetコントローラーは、スケジューリングプロファイルをサポートしていません。DaemonSetコントローラーがPodを作成すると、デフォルトのKubernetesスケジューラーがそれらのPodを配置し、DaemonSetコントローラーの`nodeAffinity`ルールに優先して従います。
```
Can you add an English ID for the target page and change to use the English ID instead?

ziyi-xie replied:
Thank you for your kind advice! I've changed to use the English ID.
And I noticed that [the original English version daemonset page ](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/#how-daemon-pods-are-scheduled) has been updated lately, and the ID used here is deleted. 
Therefore the ID I fixed here will be consistent with [the current Japanese translation of daemonset page](https://kubernetes.io/ja/docs/concepts/workloads/controllers/daemonset/#%E3%83%87%E3%83%95%E3%82%A9%E3%83%AB%E3%83%88%E3%82%B9%E3%82%B1%E3%82%B8%E3%83%A5%E3%83%BC%E3%83%A9%E3%83%BC%E3%81%AB%E3%82%88%E3%81%A3%E3%81%A6%E3%82%B9%E3%82%B1%E3%82%B8%E3%83%A5%E3%83%BC%E3%83%AA%E3%83%B3%E3%82%B0%E3%81%95%E3%82%8C%E3%82%8B%E5%A0%B4%E5%90%88), but not with the English version. 
I'll create an Issue about update the daemonset page, and will start working on it as soon as possible.

ziyi-xie replied:
Attached the Issue link here just in case. https://github.com/kubernetes/website/issues/39502

ziyi-xie replied:
@nasa9084 @t-inu @atoato88 About the ID here, the [original English page](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity-per-scheduling-profile)  (The Note part, sorry you may need to scroll down a bit.) is still using `#scheduled-by-default-scheduler` ,  but `#scheduled-by-default-scheduler`part in [DeamonSet page](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/#scheduled-by-default-scheduler) has [been deleted](https://github.com/kubernetes/website/commit/d0b3ba5ccea5138c06ece1ad9f4e4c2b0c5ef897#), so I changed to `#how-daemon-pods-are-scheduled` in the Japanese version.
Could I ask if you think this modification is appropriate or not? 
Looking forward to your comments and many thanks in advance!

nasa9084 commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -288,49 +304,29 @@ spec:
-
-上記の例では `PodAntiAffinity`を`topologyKey: "kubernetes.io/hostname"`と合わせて指定することで、redisクラスター内の2つのインスタンスが同じホストにデプロイされない場合を扱いました。
-同様の方法で、アンチアフィニティを用いて高可用性を実現したStatefulSetの使用例は[ZooKeeper tutorial](/docs/tutorials/stateful-application/zookeeper/#tolerating-node-failure)を参照してください。
+全体的な効果として、各キャッシュインスタンスは、同じNode上で実行している単一のクライアントによってアクセスされる可能性が高いです。この方法は、スキュー（負荷の偏り）とレイテンシーの両方を最小化することを目的としています。
```
```suggestion
全体的な効果として、各キャッシュインスタンスは、同じNode上で実行している単一のクライアントによってアクセスされる可能性が高いです。この方法は、スキュー(負荷の偏り)とレイテンシーの両方を最小化することを目的としています。
```

t-inu reviewed (commented):
Regarding the use of "Node" and "ノード", I'd like to discuss separately.
My review is in progress, but I left comments.

ziyi-xie commented:
> Regarding the use of "Node" and "ノード", I'd like to discuss separately. My review is in progress, but I left comments.

Thank you so much for your kind instruction! About the use of ”Node” and "ノード", maybe [this PR](https://github.com/kubernetes/website/pull/39463) is what you're referring to?

ziyi-xie commented:
/hold
Currently updating daemonset.md page. There are several [related contents](https://github.com/kubernetes/website/pull/38800#discussion_r1109169496) that need to be changed and will work on it ASAP.

t-inu commented:
> About the use of ”Node” and "ノード", maybe https://github.com/kubernetes/website/pull/39463 is what you're referring to?

Depending on what the policy in PR https://github.com/kubernetes/website/pull/39463, the translation of "node" in this page may change.
For now, I am reviewing this page without mentioning how "node" is translated.

t-inu commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -1,218 +1,237 @@
+#### Nodeアフィニティの重み
+
+`preferredDuringSchedulingIgnoredDuringExecution`アフィニティタイプの各インスタンスに、1から100の範囲の`weight`を指定できます。
+Podの他のスケジューリング要件をすべて満たすNodeを見つけると、スケジューラーはそのNodeが満たすすべての優先ルールを繰り返し実行し、対応する式の `weight`値を合計します。
```
```suggestion
Podの他のスケジューリング要件をすべて満たすNodeを見つけると、スケジューラーはそのNodeが満たすすべての優先ルールを繰り返し実行し、対応する式の`weight`値を合計に加算します。
```
> ... adds the value of the `weight` for that expression to a sum.

I think this means like "sum = sum + weight" for programming.

t-inu commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -1,218 +1,237 @@
+{{< codenew file="pods/pod-with-affinity-anti-affinity.yaml" >}}
 
-`nodeSelector`と`nodeAffinity`の両方を指定した場合、Podは**両方の**条件を満たすNodeにスケジュールされます。
+`preferredDuringSchedulingIgnoredDuringExecution`ルールにマッチするNodeがの二つ存在する場合、一つは
+`label-1:key-1`ラベル、もう一つは`label-2:key-2`ラベルで、スケジューラーは各Nodeの`weight`を考察し、その重みとNodeの他のスコアを加え、最終スコアが最も高いNodeにPodをスケジューリングします。
```
```suggestion
`preferredDuringSchedulingIgnoredDuringExecution`ルールにマッチするNodeとして、一つは`label-1:key-1`ラベル、もう一つは`label-2:key-2`ラベルの2つの候補がある場合、スケジューラーは各Nodeの`weight`を考慮し、その重みとNodeの他のスコアを加え、最終スコアが最も高いNodeにPodをスケジューリングします。
```
Suggestion for more understandable.

t-inu commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -1,218 +1,237 @@
 
-`nodeAffinity`内で複数の`nodeSelectorTerms`を指定した場合、Podは**いずれかの**`nodeSelectorTerms`を満たしたNodeへスケジュールされます。
+{{<note>}}
+この例でKubernetesにPodを正常にスケジュールさせるには、`kubernetes.io/os=linux`ラベルを持っているNodeの存在を確保しないといけません。
```
```suggestion
この例でKubernetesにPodを正常にスケジュールさせるには、`kubernetes.io/os=linux`ラベルを持つ既存のNodeが必要です。
```
> ... you must have existing nodes with the `kubernetes.io/os=linux` label.

"既存の" is better I think.

ziyi-xie commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -1,218 +1,236 @@
+`addedAffinity`はエンドユーザーには見えないので、その動作はエンドユーザーにとって予期しないものになる可能性があります。スケジューラープロファイル名と明確な相関関係のあるNodeラベルを使用すべきです。
+
+{{< note >}}
+[DaemonSetのPodを作成する](/ja/docs/concepts/workloads/controllers/daemonset/##scheduled-by-default-scheduler)DaemonSetコントローラーは、スケジューリングプロファイルをサポートしていません。DaemonSetコントローラーがPodを作成すると、デフォルトのKubernetesスケジューラーがそれらのPodを配置し、DaemonSetコントローラーの`nodeAffinity`ルールに優先して従います。
```
```suggestion
[DaemonSetのPodを作成する](/ja/docs/concepts/workloads/controllers/daemonset/#how-daemon-pods-are-scheduled)DaemonSetコントローラーは、スケジューリングプロファイルをサポートしていません。DaemonSetコントローラーがPodを作成すると、デフォルトのKubernetesスケジューラーがそれらのPodを配置し、DaemonSetコントローラーの`nodeAffinity`ルールに優先して従います。
```

ziyi-xie commented on content/ja/docs/concepts/workloads/controllers/daemonset.md:
```diff
@@ -77,7 +77,7 @@ Kubernetes1.8のように、ユーザーは`.spec.template`のラベルにマッ
 
 ## Daemon Podがどのようにスケジューリングされるか
```
```suggestion
## Daemon Podがどのようにスケジューリングされるか {#how-daemon-pods-are-scheduled}

```

ziyi-xie commented:
/unhold

atoato88 commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -1,218 +1,236 @@
+アフィニティとアンチアフィニティのメリットは以下の通りです。
 
-## 補足: ビルトインNodeラベル {#built-in-node-labels}
+* アフィニティとアンチアフィニティという用語は、より表現力が豊かです。`nodeSelector`は指定されたラベルを全て持つNodeを選択するだけです。アフィニティとアンチアフィニティは選択ロジックをより細かく制御することができます。
```
```suggestion
* アフィニティとアンチアフィニティで使われる言語は、より表現力が豊かです。`nodeSelector`は指定されたラベルを全て持つNodeを選択するだけです。アフィニティとアンチアフィニティは選択ロジックをより細かく制御することができます。
```

atoato88 commented:
The sentence `Nodeがの二つ存在する場合` is included at [this paragraph](https://deploy-preview-38800--kubernetes-io-main-staging.netlify.app/ja/docs/concepts/scheduling-eviction/assign-pod-node/#node%E3%82%A2%E3%83%95%E3%82%A3%E3%83%8B%E3%83%86%E3%82%A3%E3%81%AE%E9%87%8D%E3%81%BF) , so this is something nit.

```diff
- Nodeがの二つ存在する場合
+ Nodeが二つ存在する場合
```

atoato88 commented:
The word of `を考察し` at [this paragraph](https://deploy-preview-38800--kubernetes-io-main-staging.netlify.app/ja/docs/concepts/scheduling-eviction/assign-pod-node/#node%E3%82%A2%E3%83%95%E3%82%A3%E3%83%8B%E3%83%86%E3%82%A3%E3%81%AE%E9%87%8D%E3%81%BF) should be replaced with `を考慮し` because of preferable in this context, I think.

Plsease use the find function on the web browser, it's difficult to find out the target sentence. :sweat_smile:

atoato88 reviewed (commented):
Please see comments I left.

atoato88 commented:
To be clear current status of this PR, I commented.
It's almost LGTM.
I left something nits.

ziyi-xie commented:
> The word of `を考察し` at [this paragraph](https://deploy-preview-38800--kubernetes-io-main-staging.netlify.app/ja/docs/concepts/scheduling-eviction/assign-pod-node/#node%E3%82%A2%E3%83%95%E3%82%A3%E3%83%8B%E3%83%86%E3%82%A3%E3%81%AE%E9%87%8D%E3%81%BF) should be replaced with `を考慮し` because of preferable in this context, I think.
> 
> Plsease use the find function on the web browser, it's difficult to find out the target sentence. 😅

Thank you so much for your kind instructions! 
The sentence you mentioned here is actually currently translated as follows: 
> `preferredDuringSchedulingIgnoredDuringExecution`ルールにマッチするNodeとして、一つは`label-1:key-1`ラベル、もう一つは`label-2:key-2`ラベルの2つの候補がある場合、スケジューラーは各Nodeの`weight`を考慮し、その重みとNodeの他のスコアを加え、最終スコアが最も高いNodeにPodをスケジューリングします。
I don't know why it's not synchronized...　But the above two should be fixed already.

atoato88 commented:
Thank you for quick response.
I have confirmed that both of my suggestion are fixed already. :sweat_smile: 

I checked on [the preview site](https://deploy-preview-38800--kubernetes-io-main-staging.netlify.app/ja/docs/concepts/scheduling-eviction/assign-pod-node/), but it's not sync with latest codeset on this PR.

We should fix the error on netlify CI Job.

atoato88 commented:
[The error log](https://app.netlify.com/sites/kubernetes-io-main-staging/deploys/63fe9a0ae0c59900080cafd1) on Netlify says:
> executing "shortcodes/cve-feed.html"


I checked the codeset on this PR and found that codeset seems outdated, because following file content differ from eath other.
- ["cve-feed.html" file on your PR](https://github.com/ziyi-xie/website/blob/update-assign-pod-node/layouts/shortcodes/cve-feed.html)
- ["cve-feed.html" file on latest upstream](https://github.com/kubernetes/website/blob/main/layouts/shortcodes/cve-feed.html)

Could you rebase all files with latest upstream codeset?

ziyi-xie commented:
/hold Sorry I think I have to do some checking.

t-inu commented:
FYI: https://github.com/kubernetes/website/pull/39528#issuecomment-1449197631

atoato88 commented:
Build failure on Netlify is talked at #35780.
Same build error occured on other PR which is mentioned in that.
The root cause is not found currently, but we can try to rebuild by creating some updates on this PR.

For instance, could you update yaml file reviewed above?

atoato88 commented:
Oh, quick update!
Thank you, we can check the result on Netlify. :+1:

atoato88 commented:
Yes, we past the process on which error was occured before.
https://app.netlify.com/sites/kubernetes-io-main-staging/deploys/640598a9d86aae0007d34c34#L120

Maybe we will be able to see the demosite :pray:

ziyi-xie commented:
> Yes, we past the process on which error was occured before at. https://app.netlify.com/sites/kubernetes-io-main-staging/deploys/640598a9d86aae0007d34c34#L120
> 
> Maybe we will be able to see the demosite 🙏

Thank you so much for the kind instruction! Hope it will work this time  🙏

atoato88 commented:
Yes, we got the latest demo preview!
https://deploy-preview-38800--kubernetes-io-main-staging.netlify.app/ja/docs/concepts/scheduling-eviction/assign-pod-node/

ziyi-xie commented:
> /lgtm Finally, I can label this 👍

Thank you so much for your review and kind instructions! 
Sorry could I ask your opinion about this part?  https://github.com/kubernetes/website/pull/38800#discussion_r1125833351
It's about heading IDs. The original heading ID used in here is deleted, so it won't jump to the right place in the original English text. I changed the ID here in Japanese page, and I'm wondering if it's necessary or not. 
Looking forward to your comments and many thanks in advance!

t-inu commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -288,49 +303,29 @@ spec:
-`nodeName`はPodSpecのフィールドです。
-ここに値が設定されると、schedulerはそのPodを考慮しなくなり、その名前が付与されているNodeのkubeletはPodを稼働させようとします。
-そのため、PodSpecに`nodeName`が指定されると、上述のNodeの選択方法よりも優先されます。
+`nodeName`はアフィニティや`nodeSelector`よりも直接的なNode選択形式になります。`nodeName`はPod仕様(spec)のフィールドです。`nodeName`フィールドが空でない場合、スケジューラーはPodを考慮せずに、指定されたNodeにあるkubeletはそのNodeにPodを配置しようとします。`nodeName`を使用すると、`nodeSelector`やアフィニティおよびアンチアフィニティルールを使用するよりも優先されます。
```
```suggestion
`nodeName`はアフィニティや`nodeSelector`よりも直接的なNode選択形式になります。`nodeName`はPod仕様(spec)内のフィールドです。`nodeName`フィールドが空でない場合、スケジューラーはPodを考慮せずに、指定されたNodeにあるkubeletがそのNodeにPodを配置しようとします。`nodeName`を使用すると、`nodeSelector`やアフィニティおよびアンチアフィニティルールを使用するよりも優先されます。
```

t-inu commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -288,49 +303,29 @@ spec:
+-   指定されたNodeが存在しない場合、Podは実行されず、場合によっては自動的に削除されることがあります。
+-   指定されたNodeがPodを収容するためのリソースを持っていない場合、Podの起動は失敗し、OutOfmemoryやOutOfcpuなどの理由が表示されます。
+-   クラウド環境におけるNode名は、常に予測可能で安定したものではありません。
 
```
```suggestion

{{< note >}}
`nodeName`は、カスタムスケジューラーや、設定済みのスケジューラーをバイパスする必要がある高度なユースケースで使用することを目的としています。
スケジューラーをバイパスすると、割り当てられたNodeに過剰なPodの配置をしようとした場合には、Podの起動に失敗することがあります。
[Nodeアフィニティ](#node-affinity)または[`nodeSelector`フィールド](#nodeselector)を使用すれば、スケジューラーをバイパスせずに、特定のNodeにPodを割り当てることができます。
{{</ note >}}

```
Adding Note.
It's difficult to translate "if the assigned Nodes get oversubscribed".

t-inu commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -344,18 +339,18 @@ spec:
-[トポロジーマネージャー](/docs/tasks/administer-cluster/topology-manager/)はNodeレベルのリソース割り当てを決定する際に関与します。
+* [TaintとToleration](/ja/docs/concepts/scheduling-eviction/taint-and-toleration/)についてもっと読む。
+* [Nodeアフィニティ](https://git.k8s.io/design-proposals-archive/scheduling/nodeaffinity.md)と[Pod間アフィニティ/アンチアフィニティ](https://git.k8s.io/design-proposals-archive/scheduling/podaffinity.md)のデザインドキュメントを読む。
+* [トポロジーマネージャー](/ja/docs/tasks/administer-cluster/topology-manager/)がNodeレベルリソースの割り当て決定に参加する方法について学ぶ。
```
```suggestion
* [トポロジーマネージャー](/ja/docs/tasks/administer-cluster/topology-manager/)がNodeレベルのリソース割り当ての決定にどのように関与しているかについて学ぶ。
```

atoato88 commented:
> It's about heading IDs. The original heading ID used in here is deleted, so it won't jump to the right place in the original English text. I changed the ID here in Japanese page, and I'm wondering if it's necessary or not.

I think we can do followings:
- Merge this PR use `#scheduled-by-default-scheduler` similar with original content.
- Create a issue about "dead heading ID in `en` content", and update that by other PRs.
  - I think `en` content is basically truth even if that have some errors. that's why I wrote this step.
- We can feedback that to `ja` content by other PR.

It will take some time, but it will keep the valid of content.

ziyi-xie commented on content/ja/docs/concepts/scheduling-eviction/assign-pod-node.md:
```diff
@@ -7,212 +7,230 @@ weight: 20
+`addedAffinity`はエンドユーザーには見えないので、その動作はエンドユーザーにとって予期しないものになる可能性があります。スケジューラープロファイル名と明確な相関関係のあるNodeラベルを使用すべきです。
+
+{{< note >}}
+[DaemonSetのPodを作成する](/ja/docs/concepts/workloads/controllers/daemonset/#how-daemon-pods-are-scheduled)DaemonSetコントローラーは、スケジューリングプロファイルをサポートしていません。DaemonSetコントローラーがPodを作成すると、デフォルトのKubernetesスケジューラーがそれらのPodを配置し、DaemonSetコントローラーの`nodeAffinity`ルールに優先して従います。
```
```suggestion
[DaemonSetのPodを作成する](/ja/docs/concepts/workloads/controllers/daemonset/#scheduled-by-default-scheduler)DaemonSetコントローラーは、スケジューリングプロファイルをサポートしていません。DaemonSetコントローラーがPodを作成すると、デフォルトのKubernetesスケジューラーがそれらのPodを配置し、DaemonSetコントローラーの`nodeAffinity`ルールに優先して従います。
```

ziyi-xie commented:
> > It's about heading IDs. The original heading ID used in here is deleted, so it won't jump to the right place in the original English text. I changed the ID here in Japanese page, and I'm wondering if it's necessary or not.
> 
> I think we can do followings:
> 
> * Merge this PR use `#scheduled-by-default-scheduler` similar with original content.
> * Create a issue about "dead heading ID in `en` content", and update that by other PRs.
>   
>   * I think `en` content is basically truth even if that have some errors. that's why I wrote this step.
> * We can feedback that to `ja` content by other PR.
> 
> It will take some time, but it will keep the valid of content.

Thank you so much for the kind instructions! 
I've changed the ID in Japanese page back to  `#scheduled-by-default-scheduler` and will create an issue with `en` label to check whether it's appropriate or not.

ziyi-xie commented:
> > > It's about heading IDs. The original heading ID used in here is deleted, so it won't jump to the right place in the original English text. I changed the ID here in Japanese page, and I'm wondering if it's necessary or not.
> > 
> > 
> > I think we can do followings:
> > 
> > * Merge this PR use `#scheduled-by-default-scheduler` similar with original content.
> > * Create a issue about "dead heading ID in `en` content", and update that by other PRs.
> >   
> >   * I think `en` content is basically truth even if that have some errors. that's why I wrote this step.
> > * We can feedback that to `ja` content by other PR.
> > 
> > It will take some time, but it will keep the valid of content.
> 
> Thank you so much for the kind instructions! I've changed the ID in Japanese page back to `#scheduled-by-default-scheduler` and will create an issue with `en` label to check whether it's appropriate or not.

FYI: https://github.com/kubernetes/website/issues/39864

atoato88 commented:
/lgtm

The topic about "#scheduled-by-default-scheduler" related to: #39484

atoato88 commented:
/assign @nasa9084 
Could you review this?

ziyi-xie commented:
> /lgtm
> 
> And the topic about "#scheduled-by-default-scheduler" related to: #39484

Sorry I think the assign-pods-nodes.md file in #39484 may not be the same file as mine.

ziyi-xie commented:
/hold

ziyi-xie commented:
> > /lgtm
> > And the topic about "#scheduled-by-default-scheduler" related to: #39484
> 
> Sorry I just noticed the assign-pods-nodes.md file in #39484 may not be the same file as mine. @atoato88 @nasa9084 I'm planning to add all the heading IDs in this file ASAP, sorry for the inconvenience and could you review afterwards?

I just added all the heading IDs and could you review at your spare time please? Thanks in advance!

ziyi-xie commented:
/unhold

atoato88 commented:
/lgtm
